### PR TITLE
Add payment logging

### PR DESCRIPTION
**Pull request by maciejzgadzaj**
See https://www.drupal.org/node/2845321

### CODE REVIEW

**Comment by maciejzgadzaj** [comment]
Result:

- with full captures/refunds:

order-activity-log-full

- with partial captures/refunds:

order-activity-log

**Review by heddn — CHANGES_REQUESTED** [review]
I think the events that were added are a duplicate of the already added ones provided by state_machine. No need to add the Event and Events classes. I'll do a quick review of the rest of the patch as well.

File: modules/log/src/EventSubscriber/PaymentEventSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  }
+
+  /**
+   * Creates a log when a payment is authorized.
+   *
+   * @param \Drupal\state_machine\Event\WorkflowTransitionEvent $event
+   *   The transition event.
+   */
+  public function onAuthorizeTransition(WorkflowTransitionEvent $event) {
+    $this->dispatch(PaymentEvents::PAYMENT_AUTHORIZED, $event->getEntity());
```
Comment (heddn) [inline modules/log/src/EventSubscriber/PaymentEventSubscriber.php:59]: Shouldn't this log?

File: modules/log/src/EventSubscriber/PaymentEventSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  }
+
+  /**
+   * Creates a log when a payment is voided.
+   *
+   * @param \Drupal\state_machine\Event\WorkflowTransitionEvent $event
+   *   The transition event.
+   */
+  public function onVoidTransition(WorkflowTransitionEvent $event) {
+    $this->dispatch(PaymentEvents::PAYMENT_VOIDED, $event->getEntity());
```
Comment (heddn) [inline modules/log/src/EventSubscriber/PaymentEventSubscriber.php:69]: Why no logging?

File: modules/log/src/EventSubscriber/PaymentEventSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+  }
+
+  /**
+   * Creates a log when a payment is expired.
+   *
+   * @param \Drupal\state_machine\Event\WorkflowTransitionEvent $event
+   *   The transition event.
+   */
+  public function onExpireTransition(WorkflowTransitionEvent $event) {
+    $this->dispatch(PaymentEvents::PAYMENT_EXPIRED, $event->getEntity());
```
Comment (heddn) [inline modules/log/src/EventSubscriber/PaymentEventSubscriber.php:79]: And on through the rest of this subscriber...

File: modules/payment/src/Entity/Payment.php [hunk 4]
```diff
@@ -24,6 +24,7 @@
  *   bundle_plugin_type = "commerce_payment_type",
  *   handlers = {
  *     "access" = "Drupal\commerce_payment\PaymentAccessControlHandler",
+ *     "event" = "Drupal\commerce_payment\Event\PaymentEvent",
```
Comment (heddn) [inline modules/payment/src/Entity/Payment.php:27]: I don't think this is necessary if we state_machine

**Review by heddn — CHANGES_REQUESTED** [review]

File: modules/payment/src/Entity/Payment.php (hunk 4 above)
Comment (mglaman) [reply modules/payment/src/Entity/Payment.php:27]: It is, because payments do not use transitions.

**Comment by mglaman** [comment]
Closing for #797. Will make sure issues credits correct on D.o issue.

**Pull request closed without merging**